src/v/bytes/tests/iobuf_tests.cc

```diff
@@ -568,3 +568,31 @@ SEASTAR_THREAD_TEST_CASE(iobuf_as_istream) {
     BOOST_REQUIRE_EQUAL(underlying.size_bytes(), out.size());
     BOOST_REQUIRE_EQUAL(a + b, out);
 }
+
+SEASTAR_THREAD_TEST_CASE(iobuf_string_view_cmp) {
+    {
+        /// string_view to iobuf comparator, random string
+        auto a = random_generators::gen_alphanum_string(1024);
+        iobuf buf;
+        buf.append(a.data(), a.size());
+        BOOST_REQUIRE_EQUAL(buf, std::string_view(a));
+    }
+    {
+        /// Same length different content
+        static const ss::sstring a = "static_content";
```

Comment:
can you exercise multiple fragments. 

1. ensure the comparison will go through 2 fragments. You can do this via `prepend(temporary_buffer<char>)` of 
2. `prepend("_content")` + `prepend("static")`

I want to amke sure chunk boundaries are correct.